src/graphic/animation/nonpacked_animation.cc

```diff
@@ -0,0 +1,238 @@
+		   " frames. First image is %s",
+		   image_files.size(), playercolor_mask_image_files.size(), image_files.front().c_str());
+	}
+
+	for (const std::string& filename : image_files) {
+		const Image* image = g_gr->images().get(filename);
+		if (frames.size() && (frames.front()->width() != image->width() ||
+		                      frames.front()->height() != image->height())) {
+			throw Widelands::GameDataError(
+			   "wrong size: (%u, %u) for file %s, should be (%u, %u) like the first frame",
+			   image->width(), image->height(), filename.c_str(), frames.front()->width(),
+			   frames.front()->height());
+		}
+		frames.push_back(image);
+	}
+
+	for (const std::string& filename : playercolor_mask_image_files) {
+		// TODO(unknown): Do not load playercolor mask as opengl texture or use it as
+		//     opengl texture.
+		const Image* pc_image = g_gr->images().get(filename);
+		if (frames.front()->width() != pc_image->width() ||
+		    frames.front()->height() != pc_image->height()) {
+			throw Widelands::GameDataError("playercolor mask %s has wrong size: (%u, %u), should "
+			                               "be (%u, %u) like the animation frame",
+			                               filename.c_str(), pc_image->width(), pc_image->height(),
+			                               frames.front()->width(), frames.front()->height());
+		}
+		playercolor_mask_frames.push_back(pc_image);
+	}
+}
+
+void NonPackedAnimation::NonPackedMipMapEntry::blit(uint32_t idx,
+                                           const Rectf& source_rect,
+                                           const Rectf& destination_rect,
+                                           const RGBColor* clr,
+                                           Surface* target) const {
+	assert(!frames.empty());
+	assert(target);
+	assert(idx < frames.size());
+
+	if (!has_playercolor_masks || clr == nullptr) {
+		target->blit(destination_rect, *frames.at(idx), source_rect, 1., BlendMode::UseAlpha);
+	} else {
+		target->blit_blended(
+		   destination_rect, *frames.at(idx), *playercolor_mask_frames.at(idx), source_rect, *clr);
+	}
+}
+
+int NonPackedAnimation::NonPackedMipMapEntry::width() const {
+	return frames.at(0)->width();
+}
+int NonPackedAnimation::NonPackedMipMapEntry::height() const {
+	return frames.at(0)->height();
+}
+
+/*
+==============================================================================
+
+NonPackedAnimation IMPLEMENTATION
+
+==============================================================================
+*/
+
+NonPackedAnimation::NonPackedAnimation(const LuaTable& table, const std::string& basename)
+   : Animation(table) {
+	try {
+		// Get image files
+		if (table.has_key("pictures")) {
+			// TODO(GunChleoc): Old code - remove this option once conversion has been completed
+			mipmaps_.insert(std::make_pair(
+			   1.0f, std::unique_ptr<NonPackedMipMapEntry>(
+			            new NonPackedMipMapEntry(table.get_table("pictures")->array_entries<std::string>()))));
```

Comment:
Suggestion: Print a warning for every old-format animation if --verbose output is enabled

Reply:
Done.